packages/vite-plugin-svelte/src/preprocess.ts

```diff
@@ -76,7 +77,14 @@ function viteStyle(config: InlineConfig | ResolvedConfig = {}): {
 		const { code, map } = await transform(content, moduleId);
 
 		mapSourcesToRelative(map, moduleId);
-
+		if (map?.file === moduleId) {
+			map.file = filename;
+		}
+		if (map?.sources?.map) {
+			map.sources = map.sources.map((source: string) =>
+				source === path.basename(moduleId) ? path.basename(filename) : source
```

Comment:
both basename calls can be moved out of the map call, though i'm not sure if this is needed anymore if the code is moved above the mapSourcesToRelative call

Reply:
In my use-case the entry in sources was the absolute path that matches the moduleId.